repo2docker/buildpacks/conda/environment.py-2.7.frozen.yml

```diff
@@ -8,53 +8,50 @@ channels:
 dependencies:
   - _libgcc_mutex=0.1=conda_forge
   - _openmp_mutex=4.5=1_gnu
-  - backports=1.0=py_2
-  - backports.shutil_get_terminal_size=1.0.0=py_3
-  - backports_abc=0.5=py_1
-  - ca-certificates=2020.12.5=ha878542_0
-  - certifi=2019.11.28=py27h8c360ce_1
-  - configparser=3.7.3=py27h8c360ce_2
+  - backports=1.0=py27_1
+  - backports.shutil_get_terminal_size=1.0.0=py27_1
+  - backports_abc=0.5=py27_0
+  - ca-certificates=2021.1.19=h06a4308_0
+  - certifi=2020.6.20=pyhd3eb1b0_3
+  - configparser=4.0.2=py27_0
   - decorator=4.4.2=py_0
   - entrypoints=0.3=py27h8c360ce_1001
   - enum34=1.1.10=py27h8c360ce_1
   - futures=3.3.0=py27h8c360ce_1
   - ipykernel=4.8.2=py27_0
   - ipython=5.8.0=py27_1
-  - ipython_genutils=0.2.0=py_1
+  - ipython_genutils=0.2.0=py27_0
   - jupyter_client=5.3.4=py27_1
   - jupyter_core=4.6.3=py27h8c360ce_1
-  - ld_impl_linux-64=2.35.1=hea4e1c9_2
-  - libffi=3.2.1=he1b5a44_1007
+  - libffi=3.3=h58526e2_2
   - libgcc-ng=9.3.0=h2828fa1_18
   - libgomp=9.3.0=h2828fa1_18
   - libsodium=1.0.17=h516909a_0
   - libstdcxx-ng=9.3.0=h6de172a_18
   - ncurses=6.2=h58526e2_4
-  - openssl=1.1.1i=h7f98852_0
```

Comment:
Any idea why `openssl` is removed?

Reply:
This was a symptom of a real issue! Mamba defaults to prioritizing latest version over channel, which meant that this env was getting a lot of packages from the defaults channel (which is still updated, unlike conda-forge py27 for the most part, so it has more recent versions packaged for py2.7), whereas conda has switched strict channel priority to the default.

Setting strict channel priority explicitly results in matching behavior for mamba and conda-forge, and now openssl is installed again.

I don't know if we want to add validation to avoid this in the future? We specifically don't export envs with the channel name in them because that would prevent our `conda-forge/label/broken` strategy from working (i.e. if the same package is marked as broken, what we do now will still work, whereas specifying channel name in the frozen env would cause a package being marked as broken to prevent all future installations).